backend/database/seeding/test_helpers.go

```diff
@@ -228,22 +230,99 @@ func GetOperationFromSlug(t *testing.T, db *database.Connection, slug string) mo
 	return fullOp
 }
 
-func GetOperations(t *testing.T, db *database.Connection) []models.Operation {
-	var fullOps []models.Operation
-	err := db.Select(&fullOps, sq.Select("id", "slug", "name", "status").
-		From("operations"))
+func GetOperations(t *testing.T, db *database.Connection) []services.OperationWithID {
+	var operations []struct {
+		models.Operation
+		NumUsers    int `db:"num_users"`
+		NumEvidence int `db:"num_evidence"`
+		NumTags     int `db:"num_tags"`
+	}
+
+	var topContribs []dtos.TopContrib
+
+	var evidenceCount []dtos.EvidenceCount
+
+	err := db.Select(&operations, sq.Select("operations.id", "slug", "operations.name", "status", "count(distinct(user_operation_permissions.user_id)) AS num_users", "count(distinct(evidence.id)) AS num_evidence", "count(distinct(tags.id)) AS num_tags").
+		From("operations").
+		LeftJoin("user_operation_permissions ON user_operation_permissions.operation_id = operations.id").
+		LeftJoin("evidence ON evidence.operation_id = operations.id").
+		LeftJoin("tags ON tags.operation_id = operations.id").
+		GroupBy("operations.id").
+		OrderBy("operations.created_at DESC"))
+
 	require.NoError(t, err)
-	return fullOps
+
+	err = db.SelectRaw(&topContribs, services.GetTopContributorsForEachOperation)
+
+	require.NoError(t, err)
+
+	err = db.SelectRaw(&evidenceCount, services.EvidenceCountForAllOperations)
+
+	require.NoError(t, err)
+
+	operationsWithID := []services.OperationWithID{}
+	for _, operation := range operations {
+
+		var topContribsForOp []dtos.TopContrib
+		for i := range topContribs {
+			if topContribs[i].OperationID == operation.ID {
+				topContribsForOp = append(topContribsForOp, topContribs[i])
+			}
+		}
+
+		var evidenceCountForOp dtos.EvidenceCount
+		for i := range evidenceCount {
+			if evidenceCount[i].OperationID == operation.ID {
+				evidenceCountForOp = evidenceCount[i]
+				break
+			}
+		}
+
+		operationsWithID = append(operationsWithID, services.OperationWithID{
+			ID: operation.ID,
+			Op: &dtos.Operation{
+				Slug:          operation.Slug,
+				Name:          operation.Name,
+				Status:        operation.Status,
+				NumUsers:      operation.NumUsers,
+				NumEvidence:   operation.NumEvidence,
+				NumTags:       operation.NumTags,
+				TopContribs:   topContribsForOp,
+				EvidenceCount: evidenceCountForOp,
+			},
+		})
+	}
+	return operationsWithID
 }
 
-func GetOperationsForUser(t *testing.T, db *database.Connection, userId int64) []models.Operation {
-	var fullOps []models.Operation
-	err := db.Select(&fullOps, sq.Select("id", "slug", "name", "status").
-		From("operations").
-		LeftJoin("user_operation_permissions on operation_id = operations.id").
-		Where(sq.Eq{"user_operation_permissions.user_id": userId}))
+func GetOperationsForUser(t *testing.T, db *database.Connection, user models.User) []*dtos.Operation {
```

Comment:
Up to you, but you should be able to replace `user models.User` with `userID int64`. When you're using these functions, typically you're using a seeded user, and you should be able to just use `user.ID`, That would mean you wouldn't need to involve widdleware.UserID or the user's context.

Reply:
thanks! In this specific case I changed it to be `user` because I needed the user to create a context so I could see what operations they could view 
see `middleware.Policy(ctx).Check` on 319. But yeah, normally I'd just leave it as an ID